tools/operator/launcher/k8s/network.go

```diff
@@ -2,241 +2,522 @@ package k8s
+}
 
+func (k8s K8s) volumesList(componentType, orgName, name, dataPersistence string, enableNodeOUs bool) []v12.Volume {
+
+	volumes := make([]v12.Volume, 0)
+	volume := v12.Volume{Name: "admincerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-admincerts", orgName)}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "cacerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "cacerts", Path: fmt.Sprintf("ca.%s-cert.pem", orgName)}}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "signcerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "signcerts", Path: fmt.Sprintf("%s.%s-cert.pem", name, orgName)}}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "keystore", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "keystore", Path: "priv_sk"}}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: "tlscacerts", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "tlscacerts", Path: fmt.Sprintf("tlsca.%s-cert.pem", orgName)}}}}}
+	volumes = append(volumes, volume)
+	if enableNodeOUs {
+		volume = v12.Volume{Name: "config", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-msp", name)}, Items: []v12.KeyToPath{{Key: "config", Path: "config.yaml"}}}}}
+		volumes = append(volumes, volume)
+	}
+	volume = v12.Volume{Name: "tls", VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-tls", name)}}}}
+	volumes = append(volumes, volume)
+	volume = v12.Volume{Name: fmt.Sprintf("%s-config", componentType), VolumeSource: v12.VolumeSource{ConfigMap: &v12.ConfigMapVolumeSource{LocalObjectReference: v12.LocalObjectReference{Name: fmt.Sprintf("%s-config", name)}}}}
+	volumes = append(volumes, volume)
+	if dataPersistence == "true" || dataPersistence == "local" {
+		volume = v12.Volume{Name: "data-storage", VolumeSource: v12.VolumeSource{PersistentVolumeClaim: &v12.PersistentVolumeClaimVolumeSource{ClaimName: fmt.Sprintf("%s-data", name)}}}
+		volumes = append(volumes, volume)
 	}
-	return args
+	if componentType == "orderer" {
+		volume = v12.Volume{Name: "genesisblock", VolumeSource: v12.VolumeSource{Secret: &v12.SecretVolumeSource{SecretName: "genesisblock"}}}
+		volumes = append(volumes, volume)
+	}
+	return volumes
 }
```

Comment:
Though I think we could push some of this out into separate functions as well, we can at least improve the readability, in general we always format objects instead of leaving them as one liners when they contain more than a single element:

```
func (k8s K8s) volumesList(componentType, orgName, name, dataPersistence string, enableNodeOUs bool) []corev1.Volume {
	volumes := []corev1.Volume{
		{
			Name: "admincerts",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-admincerts", orgName),
					},
				},
			},
		},
		{
			Name: "cacerts",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-msp", name)},
					Items: []corev1.KeyToPath{
						{
							Key:  "cacerts",
							Path: fmt.Sprintf("ca.%s-cert.pem", orgName),
						},
					},
				},
			},
		},
		{
			Name: "signcerts",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-msp", name),
					},
					Items: []corev1.KeyToPath{
						{
							Key:  "signcerts",
							Path: fmt.Sprintf("%s.%s-cert.pem", name, orgName),
						},
					},
				},
			},
		},
		{
			Name: "keystore",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-msp", name),
					},
					Items: []corev1.KeyToPath{
						{
							Key:  "keystore",
							Path: "priv_sk",
						},
					},
				},
			},
		},
		{
			Name: "tlscacerts",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-msp", name),
					},
					Items: []corev1.KeyToPath{
						{
							Key:  "tlscacerts",
							Path: fmt.Sprintf("tlsca.%s-cert.pem", orgName),
						},
					},
				},
			},
		},
		{
			Name: "tls",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-tls", name),
					},
				},
			},
		},
		{
			Name: fmt.Sprintf("%s-config", componentType),
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-config", name),
					},
				},
			},
		},
	}
	if enableNodeOUs {
		volume := corev1.Volume{
			Name: "config",
			VolumeSource: corev1.VolumeSource{
				ConfigMap: &corev1.ConfigMapVolumeSource{
					LocalObjectReference: corev1.LocalObjectReference{
						Name: fmt.Sprintf("%s-msp", name),
					},
					Items: []corev1.KeyToPath{
						{
							Key:  "config",
							Path: "config.yaml",
						},
					},
				},
			},
		}
		volumes = append(volumes, volume)
	}
	if dataPersistence == "true" || dataPersistence == "local" {
		volume := corev1.Volume{
			Name: "data-storage",
			VolumeSource: corev1.VolumeSource{
				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
					ClaimName: fmt.Sprintf("%s-data", name),
				},
			},
		}
		volumes = append(volumes, volume)
	}
	if componentType == "orderer" {
		volume := corev1.Volume{
			Name: "genesisblock",
			VolumeSource: corev1.VolumeSource{
				Secret: &corev1.SecretVolumeSource{
					SecretName: "genesisblock",
				},
			},
		}
		volumes = append(volumes, volume)
	}
	return volumes
}
```

Reply:
fixed